Contributing instructions clean up (#5)
margaritahumanitarian/helpafamily-manual: contributing-to-the-repository.md

Follow these steps to sync with the helpafamily repository:

> 1. Change the directory to the new cloned one

Reviewer: ```suggestion
1.  Go into the new cloned directory:
```

   ```bash
   cd helpafamily
   ```
2. Add a remote reference to the main helpafamily repository:

   ```bash
   git remote add upstream https://github.com/margaritahumanitarian/helpafamily.git
   ```
3. Ensure the configuration looks correct:

   ```bash
   git remote -v
   ```

> ![you will have your username in place of ozer619](.gitbook/assets/sync.png)

Reviewer: ```suggestion
![You will have your username in place of ozer619](.gitbook/assets/sync.png)
```

### Running helpafamily locally

Follow these steps to run the helpafamily web app locally:

> 1. if you are not already in the cloned directory.

Reviewer: ```suggestion
1. If you are not already in the cloned directory, go into it:
```

   ```bash
   cd helpafamily/
   ```

> 2. Copy the environment variables.

Reviewer: ```suggestion
2. Copy the example *.env* file, which is where you'll put any environment variables if needed:
```

   ```bash
   cp .env.example .env
   ```

> 3. Install dependency's

Reviewer: ```suggestion
3. Install dependencies:
```

   ```bash
   yarn install
   ```

> 4. Start the server.

Reviewer: ```suggestion
4. Start the local development server:
```

Follow these steps:

> 1. Validate that you are on the `main` branch:

Reviewer: ```suggestion
1. Make sure that you are on the `main` branch:
```

   ```bash
   git status
   ```

> 2. You should get this in the command line:  

Reviewer: ```suggestion
2. You should get this at the command line:  
```

    `on branch main Your branch is up-to-date with 'origin/main'.  nothing to commit, working directory clean`

3. Sync the latest changes from the helpafamily upstream `main` branch to your local main branch:  

    ```bash
    git fetch upstream
    ```

> 4. Merges upstream main to local main.

Reviewer: ```suggestion
4. Merge upstream `main` into local `main`:
```

   ```bash
   git merge upstream/main
   ```

####    Creating a fresh branch for changes:

Working on a separate branch for each issue helps you keep your local working copy clean. You should never work on the `main`.

> 1. Create new branch and switch to it:

Reviewer: ```suggestion
1. Create a new branch and switch to it:
```

   ```bash
   git checkout -b fix_some_issue
   ```

> 2. Edit the files and once your ready to create a pull request :

Reviewer: ```suggestion
2. Edit the files. Once you're ready, create a pull request:
```

These instructions were adapted from:  

> [ https://contribute.freecodecamp.org/\#/how-to-setup-freecodecamp-locally](%20https://contribute.freecodecamp.org/#/how-to-setup-freecodecamp-locally)

Reviewer: ```suggestion
[https://contribute.freecodecamp.org/\#/how-to-setup-freecodecamp-locally](https://contribute.freecodecamp.org/#/how-to-setup-freecodecamp-locally)
```
